x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/action_log_expanded_tray.tsx

```diff
@@ -0,0 +1,221 @@
+      description: `${startedAt}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.completedAt,
+      description: `${completedAt ?? emptyValue}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.input,
+      description: `${command}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.parameters,
+      description: parametersList ? parametersList : emptyValue,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.comment,
+      description: comment ? comment : emptyValue,
+    },
+  ].map(({ title, description }) => {
+    return {
+      title: <StyledEuiCodeBlock>{title}</StyledEuiCodeBlock>,
+      description: <StyledEuiCodeBlock>{description}</StyledEuiCodeBlock>,
+    };
+  });
+
+  const getOutputContent = () => {
+    if (isExpired) {
+      return OUTPUT_MESSAGES.hasExpired(command);
+    }
+
+    if (!isCompleted) {
+      return OUTPUT_MESSAGES.isPending(command);
+    }
+
+    if (!wasSuccessful) {
+      return OUTPUT_MESSAGES.hasFailed(command);
+    }
+
+    if (isGetFileCommand) {
+      return (
+        <>
+          {OUTPUT_MESSAGES.wasSuccessful(command)}
+          <ResponseActionFileDownloadLink
+            action={action}
+            canAccessFileDownloadLink={canWriteFileOperations}
+            textSize="xs"
+            data-test-subj={getTestId('getFileDownloadLink')}
+          />
+        </>
+      );
+    }
+
+    if (isExecuteCommand) {
+      return (
+        <EuiFlexGroup direction="column" data-test-subj={getTestId('executeDetails')}>
+          {action.agents.map((agentId) => (
+            <div key={agentId}>
+              {OUTPUT_MESSAGES.wasSuccessful(command)}
+              <EuiFlexItem>
+                <ResponseActionFileDownloadLink
+                  action={action}
+                  buttonTitle={EXECUTE_FILE_LINK_TITLE}
+                  canAccessFileDownloadLink={canWriteExecuteOperations}
+                  data-test-subj={getTestId('getExecuteLink')}
+                  textSize="xs"
+                />
+              </EuiFlexItem>
+              <ExecuteActionHostResponseOutput
+                action={action}
+                agentId={agentId}
+                data-test-subj={getTestId('executeResponseOutput')}
+                textSize="xs"
+              />
+            </div>
+          ))}
+        </EuiFlexGroup>
+      );
+    }
+
+    return OUTPUT_MESSAGES.wasSuccessful(command);
+  };
+
+  const outputList = [
+    {
+      title: <StyledEuiCodeBlock>{`${OUTPUT_MESSAGES.expandSection.output}:`}</StyledEuiCodeBlock>,
+      description: (
+        // codeblock for output
+        <StyledEuiCodeBlock data-test-subj={getTestId('details-tray-output')}>
+          {getOutputContent()}
+        </StyledEuiCodeBlock>
+      ),
+    },
+  ];
+
+  return (
+    <>
+      <EuiFlexGroup
+        data-test-subj={getTestId('details-tray')}
+        direction="column"
+        style={{ maxHeight: 270, overflowY: 'auto' }}
```

Comment:
maybe memoize the styles too here since creating a new object every time will cause re-render of `EuiFlexGroup`

Reply:
7a1f87b1bee40512a7d11b44ad16cd3d781748c6